tests/test_maintel_m1m3_check_actuators.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import types
+import unittest
+
+from lsst.ts.idl.enums.MTM1M3 import BumpTest
+
+from lsst.ts.standardscripts import get_scripts_dir
+from lsst.ts.standardscripts import BaseScriptTestCase
+from lsst.ts.standardscripts.maintel.m1m3 import CheckActuators
+
+
+class TestCheckActuators(BaseScriptTestCase, unittest.IsolatedAsyncioTestCase):
+    async def basic_make_script(self, index):
+        self.script = CheckActuators(index=index, add_remotes=False)
+        self.script.mtcs.run_m1m3_actuator_bump_test = unittest.mock.AsyncMock(
+            side_effect=self.mock_test_bump
+        )
+        self.script.mtcs.enter_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.exit_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_liveliness = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_all_enabled = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_m1m3_detailed_state = unittest.mock.AsyncMock()
+
+        self.bump_test_status = types.SimpleNamespace(
+            testState=[BumpTest.NOTTESTED] * len(self.m1m3_actuator_ids)
+        )
+
+        return (self.script,)
+
+    async def mock_test_bump(self, actuator_id):
+        await asyncio.sleep(5.0)
+        self.bump_test_status.testState[actuator_id] = BumpTest.PASSED
+
+    async def test_configure_good(self):
+        """Testing a valid configuration"""
+
+        # Configure with "all" actuators
+        async with self.make_script():
+            actuators = "all"
+
+            await self.configure_script(actuators=actuators)
+
+            assert self.script.actuators == self.m1m3_actuator_ids
+
+        # Try configure with a list of valid actuators ids
+        async with self.make_script():
```

Comment:
Consider spitting each of these `async with` in its own test.. It improves the testing reporting.

e.g.:

```
        async def test_configure_all(self):

         # Configure with "all" actuators
         async with self.make_script():
             actuators = "all"

             await self.configure_script(actuators=actuators)

             assert self.script.actuators == self.m1m3_actuator_ids

    async def test_configure_valid_ids(self):
        async with self.make_script():
            actuators = [101, 210, 301, 410]

             await self.configure_script(
                 actuators=actuators,
             )

             assert self.script.actuators == actuators
```